assets/js/phoenix_live_view.js

```diff
@@ -633,7 +633,7 @@ export class LiveSocket {
         this.prevInput = input
         this.prevValue = value
         this.debounce(input, e, () => {
-          this.withinOwners(input, (view, targetCtx) => {
+          this.withinOwners(input.form, (view, targetCtx) => {
```

Comment:
Should we look at the input and then at the form? 🤔

Reply:
Hi @josevalim!

> Should we look at the input and then at the form?

Probably no. I believe the `phx_change` and `phx_submit` events should always go to the form.

This actually explains an issue I'm seeing on another project, using forms built with layers of nested components, where `handle_event` callbacks are being invoked on the wrong intermediate component for `phx_change` _because_ the form isn't being evaluated here.